src/Lean/Server/README.md

```diff
@@ -58,6 +58,7 @@ The test `tests/lean/infoTree.lean` shows how to inspect the info tree of a comm
 
 ## Code style
 
-Comments should exist to denote specifics of our implementation but, for
-the most part, we shouldn't copy comments over from the LSP specification
-to avoid unnecessary duplication.
+Comments should exist to denote specifics of our implementation but, for the most part, we shouldn't copy whole
+paragraphs of protocol description over from the LSP specification to avoid unnecessary duplication. Bug simple comments
```

Comment:
```suggestion
paragraphs of protocol description over from the LSP specification to avoid unnecessary duplication. But simple comments
```
Sounds good to me otherwise. @Vtec234 Do you agree?

Reply:
I established this style way back when because even with comments describing LSP structures, I found having the LSP spec open to the side anyway necessary for figuring out what should be sent when and in what format. In general I am very supportive of commenting everything, with this being the only exception due to the preceding reason combined with the fact that, one, copying the spec descriptions over consistently is just kind of a chore and, two, risks desync when the LSP spec does change. On the other hand if you find this genuinely useful in tooltips @lovettchris , I won't object to updating the server style.

Reply:
The problem is I get no easy intellisense and I'm worried about LSP versioning.  The comments in our code will tell us if a signature has been changed since the code was written.  I don't know how well VS code keeps documentation for past versions and would rather not depend on that.

Reply:
I see, diffing our own comments with the spec is also useful. Sounds good!